lmdeploy/model.py

```diff
@@ -210,6 +210,75 @@ def messages2prompt(self, messages, sequence_start=True):
         Returns:
             str: the concatenated prompt
         """
+
+        def _format_begin(message):
+
+            roles_cfg = dict(
+                system=dict(
+                    begin=dict(
+                        with_name='<|im_start|>system name={name}\n',
```

Comment:
It is not a common structure among all LLMs.
Please don't change the BaseChatTemplate
Try to rewrite it in derived class.